File: Prolog/eu005.pl
:- initialization(search).

search :-
    search(19, 20, Mult),
    print(Mult), nl, halt.

search(0, Acc, Acc).
search(N, Acc, Mult) :-
    lcm(N, Acc, Acc2),
    N2 is N - 1,
    search(N2, Acc2, Mult).

lcm(X, Y, Mult) :-
    XY is X * Y,
    Div is gcd(X, Y),
    Mult is XY / Div.

File: JavaScript/eu005.js
var gcd = function(a, b) {
    return (b == 0) ? a : gcd(b, a % b);
}

var lcm = function(a, b) {
    return (a * b) / gcd(a, b);
}

var result = [20,19,18,17,16,15,14,13,12,11].reduce(lcm);
console.log(result);
